### ISSUE TITLE
mupdf: EPUB is locked by DRM

### ISSUE THREAD

**Issue by nilutz** [pymupdf/PyMuPDF#1557]
## Describe the bug
While parsing EPUBs I encounter:

```
mupdf: EPUB is locked by DRM
python(81465,0x10070c580) malloc: *** error for object 0x60000261b740: pointer being freed was not allocated
python(81465,0x10070c580) malloc: *** set a breakpoint in malloc_error_break to debug
```
I guess this is a mupdf Error and therefore should not be solved within pymupdf. That being said, now it is impossible to catch this error, since it is native C failing. It would be nice to either catch these Errors and return it within Python or provide a way to find out whether this file is actually readable by mupdf.

## To Reproduce
Sorry I cannot share the respective files. But it is a "working" (and not DRM protected) epub and just called with:

```python
b = open(path, "rb").read()
file = fitz.open(path.suffix, b)
```

## Your configuration
 - macOs Monterey
 - PyMuPDF 1.19.4: Python bindings for the MuPDF 1.19.0 library.
Version date: 2022-01-01 00:00:01.
Built for Python 3.8 on darwin (64-bit).

**Comment by nilutz** [comment]
Sorry, 

after a second look - The epub seems to have a "META-INF/encryption.xml" file inside the epub zip, so it makes totally sense that this fails, due to the DRM protection. I am just confused, because I can open the file on my machine...

Sorry for the confusion, I think you can close this. Allthough it would be nice to catch these errors nonetheless.

**Comment by JorjMcKie** [comment]
Hm, actually, failures in MuPDF are turned into Python exceptions.
There is an issue with incompletely closing documents in version 1.19.4. Which wasn't there in 1.19.3 and will again be solved in v1.19.5.
So you may want to check out the behaviour in one of these cases. The new version is already available as a pre-wheel [here](https://github.com/pymupdf/PyMuPDF-wheels/tree/master/wheelhouse).
